tests/Unit/Framework/Tests/Test_PyppAnalyticSolution.cpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for detai
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <tuple>
+#include <type_traits>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/Pypp.hpp"
+#include "Framework/PyppFundamentals.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+struct AnalyticSolutionTest {
+  template <typename T>
+  struct Var1 {
+    using type = Scalar<T>;
+  };
+  template <typename T>
+  struct Var2 {
+    using type = tnsr::I<T, 3>;
+  };
+
+  AnalyticSolutionTest(const double a, const std::array<double, 3>& b)
+      : a_(a), b_(b) {}
+
+  template <typename T>
+  tuples::TaggedTuple<Var1<T>, Var2<T>> solution(
+      const tnsr::i<T, 3>& x, const double t) const noexcept {
+    auto sol =
+        make_with_value<tuples::TaggedTuple<Var1<T>, Var2<T>>>(x.get(0), 0.);
+    auto& scalar = tuples::get<Var1<T>>(sol);
+    auto& vector = tuples::get<Var2<T>>(sol);
+
+    for (size_t i = 0; i < 3; ++i) {
+      scalar.get() += x.get(i) * gsl::at(b_, i);
+      vector.get(i) = a_ * x.get(i) - gsl::at(b_, i) * t;
+    }
+    scalar.get() += a_ - t;
+    return sol;
+  }
+
+ private:
+  double a_;
+  std::array<double, 3> b_;
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Pypp.AnalyticSolution", "[Pypp][Unit]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{"Framework/Tests/"};
+  const double a = 4.3;
+  const std::array<double, 3> b{{-1.3, 5.6, -0.2}};
+  AnalyticSolutionTest solution{a, b};
+  const DataVector used_for_size(5);
```

Comment:
Good question! I think it would be possible to basically scan the function arguments passed in and try to identify something that "looks right". I think the random value code needs a bigger overhaul than what I'm giving it in this PR. There are a few ways we could combine more of the functions by using `if constexpr`, and I think it would also be really great to support the `ConversionClassList` business in `check_with_random_values`. I put in a brief effort on that front and found that basically a somewhat larger re-imagining of the implementation would be needed. Probably at the same time as that it could be revisited if there's a way to get rid of the `used_for_size`. I sort of stopped where I did in this PR because it did what I needed, and the next part required a fairly big change. I did keep the easy cleanups to `check_with_random_values` since I had them done. What do you think?

Reply:
Ah, yeah, okay -- I agree, changing everything to permit overloads or argument inference for just passing the `size_t` would need to change significantly more than the pypp changes in this PR.
Let's leave it for the future, then - - it's definitely true that putting in those alternatives will be easier with the simplified pypp code.